xml/upnp_banners.xml

```diff
@@ -541,4 +587,20 @@
     <param pos="0" name="service.product" value="XBMC"/>
     <param pos="1" name="service.version"/>
   </fingerprint>
+  <fingerprint pattern="Synology/DSM/(\d+\.\d+\.\d+\.\d+)$">
+    <description>Synology DiskStation NAS with IP</description>
+    <example host.ip="192.168.1.100">Synology/DSM/192.168.1.100</example>
```

Comment:
As a note, Synology uses a customized Linux distribution which they call DSM. That being said I don't see it being used else where except here:

https://github.com/rapid7/recog/blob/a0333ab8ab8bdecbad2bb7580b9e1d99bcaca981/xml/dns_versionbind.xml#L620-L624

Perhaps we can revisit this later and just open a ticket.